polygeist/lib/Dialect/Polygeist/Transforms/LoopInternalization.cpp

```diff
@@ -1066,6 +1067,7 @@ void LoopInternalization::runOnGPUModule(gpu::GPUModuleOp gpuModule) {
   solver.load<dataflow::SparseConstantPropagation>();
   solver.load<dataflow::IntegerRangeAnalysis>();
   solver.load<ReachingDefinitionAnalysis>(aliasAnalysis);
+  solver.load<UniformityAnalysis>(aliasAnalysis);
```

Comment:
Why do we work with the raw solver here? Can't we just query the `AnalysisManager` to give us the analyses of interest and have them setup their own solver and dependencies?

Reply:
This is customary for dataflow analysis. As an example:
![image](https://github.com/intel/llvm/assets/56368199/5085e147-04d4-4126-983d-52ce8dbf1890)